packages/volto/src/components/manage/Widgets/SchemaWidget.jsx

```diff
@@ -142,160 +172,597 @@ const makeFieldsetList = (listOfFieldsets, intl) => {
   return result;
 };
 
-/**
- * schemaField used for modal form, when editing a field
- * - based on the factory a set of fields is presented
- * - fields can be moved to another fieldset
- * @param {string} factory - the kind of field
- * @param {Object} intl
- * @param {*} fieldsets
- * @return {Object} - schema
- */
-const schemaField = (factory, intl, fieldsets) => ({
-  fieldsets: [
-    {
-      id: 'default',
-      title: 'default',
-      fields: [
-        ...['title', 'description', 'parentFieldSet'],
-        ...((factory) => {
-          switch (factory) {
-            case 'Rich Text':
-              return ['maxLength'];
-            case 'URL':
-            case 'Password':
-            case 'label_password_field':
-            case 'Email':
-            case 'label_email':
-              return ['minLength', 'maxLength'];
-            case 'Integer':
-            case 'label_integer_field':
-              return ['minimum', 'maximum'];
-            case 'Floating-point number':
-            case 'label_float_field':
-            case 'Date/Time':
-            case 'label_datetime_field':
-            case 'Date':
-            case 'label_date_field':
-            case 'File':
-            case 'File Upload':
-            case 'Image':
-            case 'Yes/No':
-            case 'label_boolean_field':
-            case 'JSONField':
-            case 'Relation Choice':
-            case 'Relation List':
-              return [];
-            case 'Multiple Choice':
-            case 'label_multi_choice_field':
-            case 'Choice':
-            case 'label_choice_field':
-              return ['values'];
-            default:
-              return ['minLength', 'maxLength'];
-          }
-        })(factory),
-        ...['required'],
-      ],
+// Register field factory properties utilities
+
+config.registerUtility({
+  name: 'Rich Text',
+  type: 'fieldFactoryProperties',
+  method: (intl) => ({
+    maxLength: {
+      type: 'integer',
+      title: intl.formatMessage(messages.maxLength),
+    },
+    default: {
+      title: intl.formatMessage(messages.defaultValue),
+      widget: 'richtext',
+      type: 'string',
     },
+  }),
+});
+
+map(
+  ['URL', 'Password', 'label_password_field', 'Email', 'label_email'],
```

Comment:
I would start using standard js for these easy things, instead of using lodash. Nowadays we don't need lodash for these simple things like map. Also, since we are not really using the output from map here, I would suggest simply doing

```javascript
['URL', 'Password', 'label_password_field', 'Email', 'label_email'].forEach
```

Reply:
Agree, fixed.